packages/shared/src/utils/validation/form-validations.ts

```diff
@@ -24,26 +31,24 @@ export const filterByLandingCompany = (landing_company, schema) =>
  * @param schema
  * @return {function(*=): {}}
  */
-export const generateValidationFunction = (landing_company, schema) => {
+export const generateValidationFunction = (landing_company: string, schema: TSchema) => {
     const rules_schema = filterByLandingCompany(landing_company, schema);
-    const rules = {};
+    const rules: { [key: string]: any } = {};
```

Comment:
I didn't have a better idea for this! so I passed any to prevent errors!

Reply:
```suggestion
    const rules: { [key: string]: string[] } = {};
```

Reply:
But the type of options is not string[][]!

Reply:
below we don't assign TOptions to keys of this local rules constant, we only assign rules from opt to it, and rules from opt are string[][] 's 🙂 thank you for applying my suggestion🙏🏻

Reply:
```suggestion
    const rules: { [key: string]: TConfig['rules'] } = {};
```
And please put the following at line 7 inside type TConfig: 
```
rules: Array<(string & TOptions)[]>;
```
working for me👍🏻